### PR TITLE
KAFKA-5384: Enable topic deletion by default

### PR DESCRIPTION
**Pull request by gwenshap**
https://cwiki.apache.org/confluence/display/KAFKA/KIP-162+-+Enable+topic+deletion+by+default

### CODE REVIEW

**Comment by ijuma** [comment]
Thanks for the PR. We should also add a note to `upgrade.html`.

**Comment by gwenshap** [comment]
@ijuma since this requires adding a new section to the upgrade notes, I'm going to assume the next release is 1.0.0.

File: docs/upgrade.html [hunk 1]
```diff
@@ -19,6 +19,17 @@
...
 <script id="upgrade-template" type="text/x-handlebars-template">
 
+<h4><a id="upgrade_1_0_0" href="#upgrade_1_0_0"</h4> Upgrading from 0.8.x, 0.9.x, 0.10.0.x, 0.10.1.x, 0.10.2 or 0.11.0.0 to 1.0.0</a></h4>
+<p>1.0.0 is fully compatible with 0.11.0.0. The upgrade can be done one broker at a time by simply bringing it down, updating the code, and restarting it.
+To upgrade from earlier versions, please review the <a href="#upgrade_11_0_0">0.11.0.0 upgrade instructions</a>.
+</p>
+
+<h5><a id="upgrade_100_notable" href="#upgrade_100_notable">Notable changes in 1.0.0</a></h5>
+<ul>
+    <li>Topic deletion is now enabled by default, since the Apache Kafka community now deems it safe to delete topics. Users who wish to
```
Comment (ijuma) [inline docs/upgrade.html]: Nit: maybe we should say that "delete topics functionality is stable" or something. Deleting topics is sort of an unsafe operation by definition. ;)

Comment (gwenshap) [reply docs/upgrade.html]: is this better?

**Review by ijuma — APPROVED** [review]
LGTM, thanks.

File: docs/upgrade.html (hunk 1 above)
Comment (ijuma) [reply docs/upgrade.html]: Looks great.

**Comment by gwenshap** [comment]
I guess you are leaving me the pleasure of merging? :)

**Comment by ijuma** [comment]
Yes!

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6150/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6134/
Test PASSed (JDK 8 and Scala 2.12).

**Pull request closed without merging**